### ISSUE TITLE
RouteController is bahaving strange when creating routes for large clusters

### ISSUE THREAD

**Issue by wojtek-t** [kubernetes/kubernetes#38618]
In large clusters, I'm seeing logs like this coming from route controller:
```
I1212 14:42:40.519542       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 1.12�s
E1212 14:42:40.569494       7 routecontroller.go:160] Could not create route ba3adfc2-c052-11e6-b5ed-42010af00002 10.162.66.0/24 for node gke-bootstrap-e2e-default-pool-0-364bdcd6-zk1z after 4.499694682s: googleapi: Error 409: The resource 'projects/kubernetes-scale/global/routes/gke-bootstrap-e2e-fefde584-ba3adfc2-c052-11e6-b5ed-42010af00002' already exists, alreadyExists
```

Logs from route controller for that node:
```
I1212 13:42:56.452677       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 3h29m54.139354633s
E1212 13:43:00.720185       7 routecontroller.go:160] Could not create route 2f7b57ea-c053-11e6-b5ed-42010af00002 10.169.95.0/24 for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 after 3h29m58.406864433s: googleapi: Error 403: Rate Limit Exceeded, rateLimitExceeded
I1212 13:43:00.720235       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 814ns
E1212 13:43:14.476678       7 routecontroller.go:160] Could not create route 2f7b57ea-c053-11e6-b5ed-42010af00002 10.169.95.0/24 for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 after 13.756433842s: googleapi: Error 403: Rate Limit Exceeded, rateLimitExceeded
I1212 13:43:14.476725       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 842ns
E1212 13:43:32.920428       7 routecontroller.go:160] Could not create route 2f7b57ea-c053-11e6-b5ed-42010af00002 10.169.95.0/24 for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 after 18.443697335s: googleapi: Error 403: Rate Limit Exceeded, rateLimitExceeded
I1212 13:43:32.920467       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 901ns
E1212 13:43:51.372235       7 routecontroller.go:160] Could not create route 2f7b57ea-c053-11e6-b5ed-42010af00002 10.169.95.0/24 for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 after 18.451756984s: googleapi: Error 403: Rate Limit Exceeded, rateLimitExceeded
I1212 13:43:51.372297       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 1.427µs
E1212 13:44:09.820327       7 routecontroller.go:160] Could not create route 2f7b57ea-c053-11e6-b5ed-42010af00002 10.169.95.0/24 for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 after 18.448025081s: googleapi: Error 403: Rate Limit Exceeded, rateLimitExceeded
I1212 14:12:06.575010       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 16m50.053648786s
E1212 14:42:32.819634       7 routecontroller.go:160] Could not create route 2f7b57ea-c053-11e6-b5ed-42010af00002 10.169.95.0/24 for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 after 47m16.298282706s: timed out waiting for the condition
I1212 14:42:32.819677       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 704ns
E1212 14:42:36.019676       7 routecontroller.go:160] Could not create route 2f7b57ea-c053-11e6-b5ed-42010af00002 10.169.95.0/24 for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 after 3.199994286s: googleapi: Error 409: The resource 'projects/kubernetes-scale/global/routes/gke-bootstrap-e2e-fefde584-2f7b57ea-c053-11e6-b5ed-42010af00002' already exists, alreadyExists
I1212 14:42:36.019734       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 1.127µs
E1212 14:42:40.519477       7 routecontroller.go:160] Could not create route 2f7b57ea-c053-11e6-b5ed-42010af00002 10.169.95.0/24 for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 after 4.499737195s: googleapi: Error 409: The resource 'projects/kubernetes-scale/global/routes/gke-bootstrap-e2e-fefde584-2f7b57ea-c053-11e6-b5ed-42010af00002' already exists, alreadyExists
I1212 14:42:40.519542       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 1.12µs
E1212 14:42:44.573306       7 routecontroller.go:160] Could not create route 2f7b57ea-c053-11e6-b5ed-42010af00002 10.169.95.0/24 for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 after 4.053757227s: googleapi: Error 409: The resource 'projects/kubernetes-scale/global/routes/gke-bootstrap-e2e-fefde584-2f7b57ea-c053-11e6-b5ed-42010af00002' already exists, alreadyExists
I1212 14:42:44.573369       7 routecontroller.go:154] Creating route for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 10.169.95.0/24 with hint 2f7b57ea-c053-11e6-b5ed-42010af00002, throttled 1.089µs
E1212 14:42:48.070154       7 routecontroller.go:160] Could not create route 2f7b57ea-c053-11e6-b5ed-42010af00002 10.169.95.0/24 for node gke-bootstrap-e2e-default-pool-3-e6254107-2ry4 after 3.496780157s: googleapi: Error 409: The resource 'projects/kubernetes-scale/global/routes/gke-bootstrap-e2e-fefde584-2f7b57ea-c053-11e6-b5ed-42010af00002' already exists, alreadyExists
```

**Comment by wojtek-t** [comment]
OK - so the "rate limit exceeded" errors were caused by colliding tests in the same project.

So the problems are that:
- timeout seems to be too low
- we are not handling errors exactly as we should

**Comment by wojtek-t** [comment]
The first one is fixed by https://github.com/kubernetes/kubernetes/pull/38619

**Comment by gmarek** [comment]
Second problem was fixed by #38701. Can we close this?

**Comment by wojtek-t** [comment]
Yes.
